src/batch/src/executor2/filter.rs

```diff
@@ -12,104 +12,88 @@
+use crate::executor2::{BoxedDataChunkStream, BoxedExecutor2, BoxedExecutor2Builder, Executor2};
 
-pub(super) struct FilterExecutor {
+pub struct FilterExecutor2 {
     expr: BoxedExpression,
-    child: BoxedExecutor,
-    chunk_builder: DataChunkBuilder,
-    last_input: Option<SlicedDataChunk>,
+    child: BoxedExecutor2,
     identity: String,
-    // FIXME: This is a quick fix as later we would use generator to limit chunk size.
-    child_can_be_nexted: bool,
 }
 
-#[async_trait::async_trait]
-impl Executor for FilterExecutor {
-    async fn open(&mut self) -> Result<()> {
-        self.child.open().await
-    }
-
-    async fn next(&mut self) -> Result<Option<DataChunk>> {
-        loop {
-            let tmp_last_input = self.last_input.take();
-
-            // We have something left from last poll of child
-            if let Some(existing_data_chunk) = tmp_last_input {
-                let (left_data_chunk, return_data_chunk) =
-                    self.chunk_builder.append_chunk(existing_data_chunk)?;
-                self.last_input = left_data_chunk;
-
-                if let Some(data_chunk) = return_data_chunk {
-                    return Ok(Some(data_chunk));
-                }
-            } else {
-                let child_input = self.fetch_one_chunk().await?;
-                if let Some(data_chunk) = child_input {
-                    self.last_input = Some(SlicedDataChunk::new_checked(data_chunk)?);
-                } else {
-                    // We should return here since nothing come from child.
-                    return if let Some(left) = self.chunk_builder.consume_all()? {
-                        Ok(Some(left))
-                    } else {
-                        Ok(None)
-                    };
-                }
-            }
-        }
-    }
-
-    async fn close(&mut self) -> Result<()> {
-        self.child.close().await
-    }
-
+impl Executor2 for FilterExecutor2 {
     fn schema(&self) -> &Schema {
         self.child.schema()
     }
 
     fn identity(&self) -> &str {
         &self.identity
     }
+
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        self.do_execute()
+    }
 }
 
-impl FilterExecutor {
-    /// Fetch one chunk from child.
-    async fn fetch_one_chunk(&mut self) -> Result<Option<DataChunk>> {
-        if self.child_can_be_nexted {
-            if let Some(data_chunk) = self.child.next().await? {
-                let data_chunk = data_chunk.compact()?;
-                let vis_array = self.expr.eval(&data_chunk)?;
-                return if let Bool(vis) = vis_array.as_ref() {
-                    let vis = vis.try_into()?;
-                    let data_chunk = data_chunk.with_visibility(vis);
-                    Ok(Some(data_chunk))
-                } else {
-                    Err(InternalError("Filter can only receive bool array".to_string()).into())
-                };
+impl FilterExecutor2 {
+    #[try_stream(boxed, ok = DataChunk, error = RwError)]
```

Comment:
Suggest using a crate-specific error type for batch executors as well, like what we did in streaming...

https://github.com/singularity-data/risingwave/blob/4798cebcead6ce6567adafc4255cf27abeb2b4bd/src/stream/src/executor_v2/error.rs#L23-L55

Reply:
Yes, will do it in following PR.